### DISCUSSION TITLE
Gcode visualisation tool

### DISCUSSION THREAD

**Discussion by Festivejelly** [kachurovskiy/nanoels#164] [Show and tell]
I built a little tool for helping to visualise Gcode for the lathe.

I took inspiration from the Lathecode tool which is great, but I wanted to be able to visualise hand crafted gcode or tweaked gcode.

Its an early version so theres likely bugs : https://festivejelly.github.io/LatheGcodeVisualiser/

It does need a bit more work but its working okay for me so far.

**Comment by kachurovskiy** [comment]
That's pretty nice, dragging that slider really helps to visualize how it will move. Would be cool to support something like this in the lathecode "Toolpath" section.

![image](https://github.com/kachurovskiy/nanoels/assets/517919/0dd9bde5-894f-485b-b77e-c8d1addfea88)

**Reply by Festivejelly** [reply]
Ive got more improvements to it too. I did this in a separate repo just so it was easier to experiment. But I can absolutely look to get that included.

**Comment by Festivejelly** [comment]
Ive tweaked the UI a little. I think it will be a useful tool to kinda hand craft or at least tweak the gcode, then be sent to the device.

But i'll do a PR to add the functionality to the Toolpath you highlighted and the larger version that appears when you click on it. I just need a little while to figure out how to integrate it into yours.

![image](https://github.com/kachurovskiy/nanoels/assets/2408274/b3ca9edf-e0bf-4f2e-b713-f7ad284ad592)

**Reply by kachurovskiy** [reply]
I'm adding basic animation to lathecode editor so the PR might be a less urgent matter. Your visualizer has more features though.

https://github.com/kachurovskiy/nanoels/assets/517919/a05c0137-3f56-47b1-a2d8-a5412369bb22

**Reply by Festivejelly** [reply]
Looks great. I store all the movements in a collection. Then the slider bar as it drags iterates over the collection and draws them onto the canvas.
